test/bpf/verifier-test.sh

```diff
@@ -61,7 +61,7 @@ function cleanup {
 }
 
 function get_section {
-	grep "__section(" $DIR/$1 | sed 's/__sec[^\"]*\"\([0-9A-Za-z_-]*\).*/\1/'
+	sed -n '/__section("/ s/.*__section("\([0-9A-Za-z_-]*\).*/\1/p' $DIR/$1
```

Comment:
> - It would collect the argument passed to __section(), even if this
  argument is not a string (but a macro, for example) and does not
  reflect the final value passed to the compiler. Address this by
  requiring an opening double quote in the pattern we use for detection.

I don't understand that part of the commit message. The double quote was already required by the sed.

I actually think it causes us to skip the templated section in `bpf_lxc`: https://github.com/cilium/cilium/blob/9d94081cef70ea8a05b32d9c81baf64ec8d6f316/bpf/bpf_lxc.c#L1412 (We don't need to fix that in this PR of course.)

Reply:
The sed requires the double quote. The grep does not. So for a `__section()` with a macro name, the whole line is taken unchanged, see:

```
$ echo '__section("foo.h")'    | grep "__section(" | sed 's/__sec[^\"]*\"\([0-9A-Za-z_-]*\).*/\1/'
foo
$ echo '__section(SOME_MACRO)' | grep "__section(" | sed 's/__sec[^\"]*\"\([0-9A-Za-z_-]*\).*/\1/'
__section(SOME_MACRO)
$ echo '__section(SOME_MACRO)' | sed -n '/__section("/ s/.*__section("\([0-9A-Za-z_-]*\).*/\1/p'
$
```

My comment is incorrect in the sense that if there's a macro instead of a string, the whole line (and not the argument passed to `__section()`) would be collected as a section name.

The `__section_tail()` is left aside by the `grep` pattern as well. The presence of the double quote in the sed expression would make the command fail to extract the correct name, not skip it.

I'll update the commit log to make it more accurate and easier to understand. I think I can also remove the `/__section("/` pattern match at the beginning of my own expression, and keep only the substitute command. This should have the same effect, but we would have just one pattern to update if we want to support the `__section_tail()` in a follow-up.